flow/flow.cpp

```diff
@@ -19,17 +19,45 @@
  */
 
 #include "flow/flow.h"
+#include "flow/DeterministicRandom.h"
 #include <stdarg.h>
 #include <cinttypes>
 
 INetwork *g_network = 0;
-IRandom *g_random = 0;
-IRandom *g_nondeterministic_random = 0;
-IRandom *g_debug_random = 0;
+
 FILE* randLog = 0;
+thread_local uint32_t deterministicRandomSeed = 0;
 uint64_t debug_lastLoadBalanceResultEndpointToken = 0;
 bool noUnseed = false;
 
+void setThreadLocalDeterministicRandomSeed(uint32_t seed) {
+	deterministicRandomSeed = seed;
+}
+
+IRandom* deterministicRandom() {
+	static thread_local IRandom* random = nullptr;
+	if(!random) {
+		if(!deterministicRandomSeed) {
+			static thread_local IRandom* preseedRandom = nullptr;
```

Comment:
Once we set `deterministicRandomSeed`, we'll never return `preseedRandom` again.  Can we just assign to `random` in `setThreadLocalDeterministicRandomSeed`, and make this just always return `random`, to avoid having the extra thread local?

Reply:
Yeah, I think that would work.

Reply:
Ok, I tried to address this in the most recent commit.